src/form_components/repeating_group.coffee

```diff
@@ -0,0 +1,155 @@
+FormRenderer.Models.ResponseFieldRepeatingGroup = FormRenderer.Models.BaseFormComponent.extend
+  group: true
+  field_type: 'repeating_group'
+
+  initialize: ->
+    FormRenderer.Models.BaseFormComponent::initialize.apply @, arguments
+    @entries = []
+
+  validateComponent: ->
+    for entry in @entries
+      entry.formComponents.invoke('validateComponent')
+
+  setExistingValue: (entryValues) ->
+    if @isRequired()
+      # If the field is required, ensure that there is at least one value present
+      if !entryValues || entryValues.length == 0
+        entryValues = [{}]
+
+    else # Field is optional...
+      # If entryValues is not set, then add an empty value.
+      if !entryValues
```

Comment:
The comment below better describes the difference between `[]` (skipped) and `[{}]` (not skipped, but not filled out yet) values here, but the logic on line 16 (`if !entryValues || entryValues.length == 0`) encapsulates both skipped _and_ not-filled-out values, rather than handling either separately, since in both cases on a required field we want to create an initial entryValue to populate nested fields into.

So, for cases where a field is _optional_:

If a user doesn't answer a field,

* If we render the form without a response (so either a new form or a saved form with no answer from a respondent for this field), `entryValues` will be undefined (since the respondent didn't answer it) and `!entryValues` is true
* If a user then makes the field required, re-rendering the form will force the first nested group to appear, since `!entryValues` will still be true.

If a user skips a field,

* If a respondent skips an optional field, it will save/serialize to `[]`
* If the user then changes the field to required, re-rendering the form will evaluate `entryValues.length == 0` to true (since `entryValues` will be `[]`) for the same outcome as the above: forcing the first nested group to appear so the user has to answer it.

This is all super dense and not super intuitive, so feel free to ping me for a quick call or chat on Slack if anything's still unclear.